src/middlewared/middlewared/plugins/cron.py

```diff
@@ -195,3 +198,76 @@ def cron_extend(self, data):
         await self.middleware.call('service.restart', 'cron')
 
         return response
+
+    @accepts(
+        Int('id')
+    )
+    @job(lock=lambda args: args[-1], logs=True)
+    def run(self, job, id):
+        """
+        Job to run cronjob task of `id`.
+        """
+        def __cron_log(line):
+            job.logs_fd.write(line)
+            syslog.syslog(syslog.LOG_INFO, line.decode())
+
+        cron_task = self.middleware.call_sync('cronjob._get_instance', id)
+        cron_cmd = ' '.join(
+            self.middleware.call_sync(
+                'cronjob.construct_cron_command', cron_task['schedule'], cron_task['user'],
+                cron_task['command'], cron_task['stdout'], cron_task['stderr']
+            )[7:]
+        )
+
+        job.set_progress(
+            10,
+            'Executing Cron Task'
+        )
+
+        syslog.openlog('cron', facility=syslog.LOG_CRON)
```

Comment:
You're opening/closing log in middlewared process but rsync/cron commands run in another process, isnt that wrong/weird?

Reply:
So for rsync, syslog is not happening at all - we are only logging to the jail log which i think is sufficient. For cron, i don't think it's wrong. Callback is being run in middlewared process and syslog is also open in that one. Please let me know if i am missing something 🤔